Lib/test/test_future.py

```diff
@@ -250,11 +250,21 @@ def test_annotations(self):
         eq('str or None if sys.version_info[0] > (3,) else str or bytes or None')
         eq("f'f-string without formatted values is just a string'")
         eq("f'{{NOT a formatted value}}'")
-        eq("f'some f-string with {a} {few():.2f} {formatted.values!r}'")
-        eq('''f"{f'{nested} inner'} outer"''')
-        eq("f'space between opening braces: { {a for a in (1, 2, 3)}}'")
-        eq("f'{(lambda x: x)}'")
-        eq("f'{(None if a else lambda x: x)}'")
+        eq("f'some f-string with {a!f} {few()!f:.2f} {formatted.values!r}'")
```

Comment:
Why these tests were changed? Can you preserve the existing tests and add new tests?

Reply:
I'll double check. These might (or might not) be left over from the !d version. The ast unparser code has been through a number of revisions.

Reply:
They were changed because at one point the AST didn't have enough information to know whether !f had been used in the original source, so the unparser always added !f (since it was the default). This is another case where the exact fidelity of the annotations unparser is an issue.

But in the current implementation, the !f isn't added unless it's explicitly provided, so I can undo these changes.

Reply:
I restored these. Thanks for catching it.